BrightFutures/ExecutionContext.swift

```diff
@@ -50,9 +50,22 @@ public func toContext(queue: dispatch_queue_t) -> ExecutionContext {
     return Queue(queue: queue).context
 }
 
-typealias ThreadingModel = () -> ExecutionContext
+/// Defines BrightFutures' default threading behaviour, by default this is
+/// set to `StandardThreadingModel.AsynchronousOnCurrentThread`
+public var DefaultThreadingModel: StandardThreadingModel = .AsynchronousOnCurrentThread
```

Comment:
I'm not sure exactly what your use case is, but I suspect this could be done by writing your own extension or helper methods, without necessarily needing to be added to BrightFutures as a whole. This PR doesn't restrict what users can do, it just puts a firm barrier in place which prevents the default threading model from being changed arbitrarily from any part of the code (which is a dangerous thing to allow people to do). If users want a more complicated threading model they can implement ExecutionContexts and inject them as necessary.

Anyhow, it's Christmas, so I'm getting off GitHub. Hope you're all having a good day! :smile: